classes/Product.php

```diff
@@ -4296,12 +4296,16 @@ public function checkDefaultAttributes()
 
     /**
      * @param array $products
-     * @param bool $have_stock
+     * @param bool|null $have_stock DEPRECATED
      *
      * @return array|false
      */
-    public static function getAttributesColorList(array $products, $have_stock = true)
+    public static function getAttributesColorList(array $products, $have_stock = null)
```

Comment:
```suggestion
    public static function getAttributesColorList(array $products, $have_stock = true)
```

Isn't a BC break, because if someone overwrites it and use `$have_stock` the result will be different :thinking:

Reply:
this parameter is not used inside a method, right?

Reply:
Yes, but doesn't mean modules are not using it :(

Reply:
@PierreRambaud 

Ok i put again true and i check if it's !== true ?

Because at the start of discussion ?

`It will be not better to change $have_stock with null default value and use the displayParameterAsDeprecated('have_stock'); ?`

Reply:
@okom3pom if we want to keep the same signature indeed it's the right call